src/semidiscretization/semidiscretization_hyperbolic.jl

```diff
@@ -135,39 +135,48 @@ function Base.show(io::IO, ::MIME"text/plain", semi::SemidiscretizationHyperboli
     summary_line(io, "mesh", semi.mesh)
     summary_line(io, "equations", semi.equations |> typeof |> nameof)
     summary_line(io, "initial condition", semi.initial_condition)
-    if semi.boundary_conditions isa UnstructuredQuadSortedBoundaryTypes
-      @unpack boundary_dictionary = semi.boundary_conditions
-      summary_line(io, "boundary conditions", length(boundary_dictionary))
-      for (boundary_name, boundary_condition) in boundary_dictionary
-        summary_line(increment_indent(io), boundary_name, typeof(boundary_condition))
-      end
-    else # non dictionary boundary conditions container
-      summary_line(io, "boundary conditions", 2*ndims(semi))
-      if (semi.boundary_conditions isa Tuple ||
-          semi.boundary_conditions isa NamedTuple ||
-          semi.boundary_conditions isa AbstractArray)
-        bcs = semi.boundary_conditions
-      else
-        bcs = collect(semi.boundary_conditions for _ in 1:(2*ndims(semi)))
-      end
-      summary_line(increment_indent(io), "negative x", bcs[1])
-      summary_line(increment_indent(io), "positive x", bcs[2])
-      if ndims(semi) > 1
-        summary_line(increment_indent(io), "negative y", bcs[3])
-        summary_line(increment_indent(io), "positive y", bcs[4])
-      end
-      if ndims(semi) > 2
-        summary_line(increment_indent(io), "negative z", bcs[5])
-        summary_line(increment_indent(io), "positive z", bcs[6])
-      end
-    end
+
+    print_boundary_conditions(io, semi)
+
     summary_line(io, "source terms", semi.source_terms)
     summary_line(io, "solver", semi.solver |> typeof |> nameof)
     summary_line(io, "total #DOFs", ndofs(semi))
     summary_footer(io)
   end
 end
 
+# type alias for dispatch in printing of boundary conditions
+const SemiHypMeshBCSolver{Mesh,BoundaryConditions,Solver} = 
+      SemidiscretizationHyperbolic{Mesh, Equations, InitialCondition, BoundaryConditions, 
+                                   SourceTerms, Solver} where {Equations, InitialCondition, SourceTerms}
+
+# generic fallback: print the type of semi.boundary_condition.
+print_boundary_conditions(io, semi::SemiHypMeshBCSolver) = summary_line(io,"boundary conditions", typeof(semi.boundary_conditions))
+
+function print_boundary_conditions(io, semi::SemiHypMeshBCSolver{<:AbstractMesh,<:UnstructuredQuadSortedBoundaryTypes})
+  @unpack boundary_conditions = semi
+  @unpack boundary_dictionary = boundary_conditions
+  summary_line(io, "boundary conditions", length(boundary_dictionary))
+  for (boundary_name, boundary_condition) in boundary_dictionary
+    summary_line(increment_indent(io), boundary_name, typeof(boundary_condition))
+  end  
+end
+
+function print_boundary_conditions(io, semi::SemiHypMeshBCSolver{<:AbstractMesh,<:Union{Tuple,NamedTuple,AbstractArray}})
```

Comment:
```suggestion
function print_boundary_conditions(io, semi::SemiHypMeshBCSolver{<:AbstractMesh, <:Union{Tuple,NamedTuple,AbstractArray}})
```
Do you use `AbstractArray`? If not, is there any setup using that? If not, I would propose to remove `AbstractArray` here.

Reply:
I don't, but it was included in the original `Base.show` implementation. I left it there so that it would reproduce the original behavior.